accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,126 @@
+
+Source generator conflict resolution should run before compile, but after all NuGet package assets have been evaluated.  Specific sequencing, public/private target naming, extensibility will not be specified here but should follow best practice as determined by the SDK team. 
+
+### Source generator versioning
+
+Source generators should be strongly named and update assembly version with every public stable release (GA and servicing).  This ensures that the publicly shipped source generators can be loaded side-by-side in the same process and garuntees 
+
+### Source generator compatibility concerns
+
+Source generators should only use public API in framework components they extend.  This public API already needs to meet the strict binary compatibility guarantees as framework API.  As such libraries containing source-generated code will be binary-compatible with future releases, just as if the user wrote this code directly.
+
+Source generators should not use public API with "reserved" or "undocumented" functionality in attempts to avoid public API guarantees.  Libraries paired with source generators need to provide the same guarantees as libraries alone do today.
+
+Source generators should avoid emitting public API in user's projects.  Such API can create a compatibility contract in that user's library which needs to be maintained.  If a source generator produces public API in a user's project by design, that public API must be strictly maintained following the same rules as framework components across versions of the source generator.  Care should be taken to the public API which is generated and should be reviewed by the Framework Design Council.  A similar constraint applies to internal API: users will have source that expects to interact with the generated internal API and the user will expect that code to continue to compile between versions.
+
+Generated code should not be generated in System namespaces, it should prefer the user's namespace.  Generated code should be triple-slashed commented and be free from compiler warnings.  Generated code should avoid name colisions with user code and provide affordances for resolving name conflicts. 
+
+### Functionality concerns for source generators
+
+Though the following are not specific to inbox source generators, these concerns are worth reviewing to describe how they apply to framework specific source generators.
+
+Source generators target .NETStandard and must run on both .NETFramework and .NETCore.  
```

Comment:
> Source generators target .NETStandard

Meaning the code in the source generator itself, not the code it generates.

Reply:
Correct.  Will clarify.

Reply:
When determining what the target runtimes are of the generated code, the generator examines the types available in the references passed in, right? As the compiler does to light up some features. It doesn't look at the TFM.

Reply:
It is up to the each source generator whether to light-up or not. I expect that most of the inbox source generators will be hardcoded to latest runtime version, and won't do any light up.